Translate Tips & FAQ section to English (#426)
evcc-io/docs: i18n/en/docusaurus-plugin-content-docs/current/guides/setup.mdx

---
sidebar_position: 2
---

# Setup & Configuration

## Configuration

### A note on individual tastes

::::caution
evcc has a lot of settings, knobs, and dials. The default parameters are based on lots of experience from many users, and most of the time, the default should work well for you.

**If you do start making changes, remember the golden rule:**

:::danger
**Since parameters sometimes influence each other, it is STRONGLY encouraged to only make changes to one setting at at time, and test the effects of each change before making further modifications.**

> :::

Reviewer: I wont do the nested blocks here. I think CAUTION+DANGER is a little much :D

Author: Think I agree here, but I wanna check on a laptop to see what looks best, gimme a bit 👀

Author: Resolved in [af64646](https://github.com/evcc-io/docs/pull/426/commits/af646464ccfae587ce7dffba35428f2c5138968e)

::::

### Syntax Error in evcc.yaml

Yaml is very sensitive to syntax - and errors don't always catch the eye straight away. Linters such as (https://onlineyamltools.com/validate-yaml) can be super useful, and are worth checking to find simple mistakes.

## Common Problems

### Something's not working. What now?

We have a [Community Support](https://github.com/evcc-io/evcc/discussions) forum, with developers and users standing by to help solve common issues.

When making a post, it would be super useful to give as much information as possible on the following:

- As precise an explanation as possible as to what your problem is

- Which devices (vehicles, meters, chargers) are in use?

- Your full, sanitised `evcc.yaml`

<details>
<summary>
   Helper comamnd (*please make sure all usernames, passwords, and other secrets are removed!*).
</summary>

`egrep -v '^#|^\s+#' evcc.yaml |grep . |sed -E 's/(password|user|token|url|host|broker).*/\1: *****/'`

</details>

  - Alternatively, you can use `evcc dump --cfg >~/config.txt` to create an anonymized configuration file at `~/config.txt`.

**⛑️ First Aid**

> We've added a useful helper utility to evcc to help with the above. By entering `evcc discuss` in your terminal, our helper wizard will help you create a new discussion topic in our "First Aid" section.

Reviewer: The topic in GitHub discussions is not localized and GH gives us no option to add localization. So we should keep Erste Hilfe" when describing the navigation. At least for now. Maybe add a notice that it's totally fine to also post in English.

Renaming the GH topic to English is something we should do eventually.

```suggestion
We've added a useful helper utility to evcc to help with the above. By entering `evcc discuss` in your terminal, our helper wizard will help you create a new discussion topic in our "Erste Hilfe" section.
```

Author: Concur, should we make this read `"Erste Hilfe" (first aid)` to make it nice and clear what the section means?

Reviewer: Good point. I changed it.

This discussion will automatically include any relevant error messages, and the anonymized configuration file - you only need to supply a meaningful title and a description of your problem.

This First Aid utility can only be use for the initial creation of a discussion thread. If you need to supply further information to a topic already in progress, you'll need to do this manually.

> ### How do I create a log fire for error analysis?

Reviewer: ```suggestion
### How do I create a log file for error analysis?
```
🪵🔥😄

In its standard configuration, evcc only issues log messages if an _error_ occurs (see also the [`log`](docs/reference/configuration/log#log) configuration key).

When evcc is started directly from the console, log messages are printed directly to the screen (`STDOUT`).

You can test out different parts of your installation using `evcc charger`, `evcc meter`, and `evcc vehicle`. Note that the log level defined in your `evcc.yaml` is used with these commands, but you can override it if necessary:

`Example: evcc -l debug meter`

If evcc is started as a Linux system service (as per [Linux Autostart](/docs/installation/manual#autostart)), you should be able to find your logs using the following commands:

- Follow the log in real time

>   `sudo journalctl -fau evcc`

Reviewer: Line wraps don't work. Not quite sure why.

![Bildschirmfoto 2023-08-22 um 15 34 56](https://github.com/evcc-io/docs/assets/152287/47894446-1337-4b5b-b643-7607dac83042)

Author: Probably worth investigating why this is happening with the element, might need to set a wrapping CSS rule on inline code elements

For now, resolved in [988f08e](https://github.com/evcc-io/docs/pull/426/commits/988f08e72acab3818a800f013e87c923614d40d5)

- Show the log since the last start of the evcc service (exit with Ctrl+C)
  `sudo journalctl -u evcc -q`
- Save the above log to a file in the home directory
  `sudo journalctl -u evcc -q > ~/evcc.log`
- You can also define a Start (`-s`) and End (`-e`) timestamp:
  `sudo journalctl -u evcc -S "2023-03-21 07:00:00" -U "2023-03-21 08:00:00" -q > ~/evcc.log`

You might find more useful commands here: [wiki.archlinux.org/title/Systemd/Journal](https://wiki.archlinux.org/title/Systemd/Journal#Filtering_output)

If you're using Docker, you should be able to use `docker logs`. See the [Docker documentation](https://docs.docker.com/config/containers/logging/) for more details.

### More thoughts on device detection

`evcc detect` is a special command that attempts to find compatible hardware on your network.
In particular, it can sometimes help find "new" Sunspec-compatible modbus devices - however, it is more of a developer / support tool for diagnostic purposes, and can't provide detailed results.

> ## Common Errors

Reviewer: Sorry. I wasn't clear in expressing what I thought :) Yes, we keep this change and update DE afterwards.

You can use a program such as `htop` to help you diagnose whether another instance of evcc is running in the background.

If you do have a reason to use evcc at the terminal, make sure to stop the service (for example, with `systemctl stop evcc`) beforehand.

## Telemetry & Community Data

The [evcc Website](https://evcc.io/#live) (and the "Charge Energy Overview" dialog in the evcc UI) shows aggregated live charging data from evcc installations. We collect this data on our central _api.evcc.io_ server - participation is completely voluntary.

### How do I participate?

Simply turn on the toggle in the "Charge Energy Overview" dialog in the evcc UI.

You can also activate this using the [`telemetry: true`](/docs/reference/configuration/telemetry/) key in the configuration, but this is a deprecated method and may be removed at a later date.

**A 💚 Sponsor Token is currently required to participate in Community Data**. This helps ensure that our data quality stays high, and poor / fake data stays out.

### What data is currently being collected?

We currently collect the following:

- current charging power

> - current proportion of charging power supplied by Solar

Reviewer: ```suggestion
- current proportion of charging power supplied by solar
```

- total charged energy

> - total proportion of energy supplied by Solar

Reviewer: ```suggestion
- total proportion of energy supplied by solar
```

**Calculation of savings / effective price**

The algorithm distinguishes between grid supply and self-generated solar energy (solar and house battery).

The cost advantage of your solar energy is calculated from the difference between your grid import rate (e.g €0.294/kWh) and your feed-in tariff (e.g €0.08/kWh).
In this example, each unit of produced energy is €0.214 (0.294 - 0.08) cheaper than the grid import rate.
If you charged a vehicle with 10 kWh of your own energy, this would then correspond to an effective saving of €2.14.

If you charged 100% with your own solar energy, the displayed _effective energy price_ would be the cost of not exporting to the grid, i.e the feed-in tariff (€0.08/kWh).
If you charge with 50% solar energy and 50% grid power, the _effective energy price_ adapts accordingly (e.g €0.187/kWh).

If you don't receive a feed-in tariff for exporting power to the grid, you can set the feed-in price to 0 - the solar energy is then treated as being free of charge.
If you have to pay tax on self-used energy, you can set the rate on your feed-in rate to get more realistic numbers.

**Calculation of the solar energy share**

> If you get energy from more than one source at the same time (e.g 50% grid, 50% Solar), the algorithm will distribute the sources evenly across all configured Chargers.

Reviewer: ```suggestion
If you get energy from more than one source at the same time (e.g 50% grid, 50% solar), the algorithm will distribute the sources evenly across all configured Chargers.
```
